plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -0,0 +1,185 @@
+    this(new GoogleApiUrls());
+  }
+
+  @VisibleForTesting
+  GoogleApiFactory(GoogleApiUrls googleApiUrls) {
+    this.googleApiUrls = googleApiUrls;
+  }
+
+  @Activate
+  public void init() {
+    Preconditions.checkNotNull(proxyService, "proxyService is null");
+    jsonFactory = new JacksonFactory();
+    buildTransports();
+    proxyService.addProxyChangeListener(new IProxyChangeListener() {
+      @Override
+      public void proxyInfoChanged(IProxyChangeEvent event) {
+        buildTransports();
+      }
+    });
+  }
+
+  /**
+   * @return the CloudResourceManager/Projects API
+   */
+  @Override
+  public Projects newProjectsApi(Credential credential) {
+    synchronized (transports) {
+      checkFields();
+      CloudResourceManager resourceManager =
+          new CloudResourceManager.Builder(transports.get(googleApiUrls.cloudResourceManagerUrl()),
+                                           jsonFactory, credential)
+          .setApplicationName(CloudToolsInfo.USER_AGENT).build();
+      Projects projects = resourceManager.projects();
+      return projects;
+    }
+  }
+
+  /**
+   * @return the Appengine/Apps API
+   */
+  @Override
+  public Apps newAppsApi(Credential credential) {
+    synchronized (transports) {
+      checkFields();
+      Appengine appengine =
+          new Appengine.Builder(transports.get(googleApiUrls.appEngineAdminUrl()), jsonFactory,
+                                credential)
+          .setApplicationName(CloudToolsInfo.USER_AGENT).build();
+      Apps apps = appengine.apps();
+      return apps;
+    }
+  }
+
+  private void checkFields() {
+    Preconditions.checkNotNull(transports, "transport is null");
+    Preconditions.checkNotNull(jsonFactory, "jsonFactory is null");
+  }
+
+  private void buildTransports() {
+    synchronized (transports) {
+      try {
+        buildAppEngineAdminTransport();
+        buildCloudResourceManagerTransport();
+      } catch (URISyntaxException ex) {
+        logger.log(Level.SEVERE, "Could not create transport using the proxy settings", ex);
+        transports = null;
+      }
+    }
+  }
+
+  private void buildAppEngineAdminTransport() throws URISyntaxException {
+    buildTransportForUrl(googleApiUrls.appEngineAdminUrl());
+  }
+
+  private void buildCloudResourceManagerTransport() throws URISyntaxException {
+    buildTransportForUrl(googleApiUrls.cloudResourceManagerUrl());
+  }
+
+  private void buildTransportForUrl(String url) throws URISyntaxException {
+    ConnectionFactory connectionFactory =
+        new TimeoutAwareConnectionFactory(createProxy(url), DEFAULT_TIMEOUT_MS, DEFAULT_TIMEOUT_MS);
+    transports.put(url,
+                   new NetHttpTransport.Builder().setConnectionFactory(connectionFactory).build());
+  }
+
+  private Proxy createProxy(String url) throws URISyntaxException {
+    Preconditions.checkNotNull(proxyService, "proxyService is null");
+    Preconditions.checkNotNull(googleApiUrls, "googleApiUrl is null");
+
```

Comment:
Since none of the Google APIs support HTTP anymore (right?) then we could assert here that the URL does not start with 'http://', so we should never see a proxy-data of type HTTP.